Docker compose + streamable http

ividal opened the pull request:
This is in draft mode and will be parked until either: 
- any-agent supports streamable http (the new standard, this is ongoing work)
- we decide to keep stdio for now when deploying.

We get a docker compose file that starts the bot app and, separately, the MCP server. 
The MCP server container will have a port exposed for http requests.
Caveat: the code still has the MCPSse, which will likely change soon.

It requires MCP_DISCORD=http://mcp-discord:8080/mcp as an env var.

Closes #12

macaab26 committed:
Changing did

ividal committed:
chore: cleaner passing the url of the server

ividal committed:
chore: docker compose. Caveat: it assumes comm between bot and MCP server works (SSE broken for now)

macaab26 commented on Makefile:
```diff
@@ -24,4 +24,12 @@ test: build
+
+# 'build' is a prerequisite for 'up', so the image is always built before starting services
+up: build
+	docker-compose up -d
```
Change this for `docker compose` please, with that you don't have to install the Docker plugin, works with the Docker binary

ividal replied:
Done!

macaab26 commented on odinbot/agent.py:
```diff
@@ -182,33 +182,17 @@ async def _create_agent(self) -> AnyAgent:
         )
         
+        MCP_DISCORD_URL = os.environ.get('MCP_DISCORD', 'http://localhost:8080/mcp')
+        logger.info(f"{MCP_DISCORD_URL=}")
```
Besides the logger, this is exactly my PR, I think it would be better if we merge it into this branch, and we close it, WDYT?

ividal replied:
Do you mean [this PR](https://github.com/mozilla-ai/0din-bot/pull/8/files#diff-83943c92fdd9f7204ea0b4d6d06a6f2bfd2f4a07d52474c850b532ad9c363fc3R191) or the [test Dockerfile PR](https://github.com/mozilla-ai/0din-bot/pull/17)? 

- For PR#8 (the switch to SSE, which we now know was deprecated), I agree it should be closed. And I should modify this one back to MCPStdio if we want to merge the docker compose stuff into main. I'll do that right now.

- If you mean PR#17 (the [test Dockerfile PR](https://github.com/mozilla-ai/0din-bot/pull/17) ), let me know what you prefer. In this PR we have a single Dockerfile (causing a docker image about 1MB bigger than if we didn't include test deps). But in general I'd agree with keeping prod/test images separate, so I'm also OK with having 2 separate Dockerfiles. I'd only ask that we don't have to install the system twice, or we'll diverge very quickly and we may end up not testing what we thought we were. WDYT?

ividal replied:
Second thoughts on the PR#8 point: if I revert to MCPStdio, this whole PR makes no sense, because the two containers would be separate and would not talk Stdio. So I'd rather address whatever we want from the docker side of the PR, but not merge anything until we have an MCPStreamableHttp on any-agent to replace the MCPSse lines.

macaab26 replied:
I'm cool with that 😄 . Will close my [PR](https://github.com/mozilla-ai/0din-bot/pull/8)

ividal committed:
chore: manual test of mcp server, streamable http

ividal committed:
chore: separate stages (test, prod) in Dockerfile

ividal committed:
chore: add odinbot.tools to path

ividal committed:
chore: single stage

ividal committed:
chore: docker compose, no dash

ividal commented on odinbot/agent.py:
```diff
@@ -182,33 +182,17 @@ async def _create_agent(self) -> AnyAgent:
-                        ],
-                        client_session_timeout_seconds=60.0,  # Increased timeout to 60 seconds
-                    ),
+                    MCPSse(
```
TODO before merging (and why this is a draft):
```suggestion
                    MCPStreamableHttp(
```

Pending [this nice update](https://github.com/mozilla-ai/any-agent/issues/488).

ividal replied:
Aaand there we are. Any Agent supports streamable http now, so we can do this.

ividal committed:
feature(agent): use streamable http with discord mcp server

feature(agent): agent uses Discord MCP server via streamable http + docker compose

ividal committed:
chore: use original mcp server docker image

ividal committed:
doc: docker compose in README

ividal committed:
doc: readme
